default_app/main.js

```diff
@@ -234,40 +234,62 @@ function loadApplicationPackage (packagePath) {
     packagePath = path.resolve(packagePath)
     const packageJsonPath = path.join(packagePath, 'package.json')
     if (fs.existsSync(packageJsonPath)) {
-      const packageJson = JSON.parse(fs.readFileSync(packageJsonPath))
+      let packageJson
+      try {
+        packageJson = JSON.parse(fs.readFileSync(packageJsonPath))
+      } catch (e) {
+        showErrorMessage('Unable to parse package.json, it contains errors.\n\n' +
+          `${e.toString()} in ${packageJsonPath}`)
```

Comment:
I don't think we should say `it contains errors` here since the error caught here could be thrown from both `JSON.parse` and `fs.readFileSync` so it might have failed to read but actually be valid JSON.